ntoskrnl/se/token.c

```diff
@@ -4303,7 +4303,7 @@ SepAdjustPrivileges(
                     /* Copy SE_PRIVILEGE_ENABLED | SE_PRIVILEGE_REMOVED */
                     NewAttributes = NewState[j].Attributes;
                     NewAttributes &= (SE_PRIVILEGE_ENABLED | SE_PRIVILEGE_REMOVED);
-                    NewAttributes |= Token->Privileges[i].Attributes & ~SE_PRIVILEGE_ENABLED;
+                    NewAttributes |= Token->Privileges[i].Attributes & ~(SE_PRIVILEGE_ENABLED | SE_PRIVILEGE_ENABLED_BY_DEFAULT);
```

Comment:
What happens to the test results if you don't modify this line (and the other modified one below) ?
I'm not convinced this is a correct patch. This enabled_by_default flag acts like an informative flag for telling that said privilege, by default, is enabled. But this privilege can have also been disabled manually by someone, and the remaining presence of the enabled_by_default flag should not influence in any case the detection of whether a privilege is currently enabled or disabled.

Reply:
Ok, it looks like ReactOS can clearly disable privileges enabled by default without problems even without these lines changed. Guess I have had a brain fart moment misunderstanding that SE_PRIVILEGE_ENABLED_BY_DEFAULT is also an indicator alongside with SE_PRIVILEGE_ENABLED where a privilege is disabled or enabled in the kernel. I will refactor this PR and only provide a testcase at this point and rethink the loops in #3938.